pkg/component/component.go

```diff
@@ -640,6 +641,21 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 		return errors.New(fmt.Sprintf("Directory / file %s is empty", path))
 	}
 
+	// FIXME(alanconway) earlier setup?
+
+	// Start reading local files concurrently, no changes yet.
+	globExps = append(globExps, ".odo", ".git", ".odoignore", ".gitignore")
```

Comment:
we are already adding ignore rules here https://github.com/openshift/odo/blob/a6970c43592670ea7a1d59627166a168a3fd03a6/pkg/util/util.go#L423

Maybe add these rules there itself?